modules/alloy/src/main/ts/ephox/alloy/behaviour/highlighting/HighlightApis.ts

```diff
@@ -75,18 +75,18 @@ const isHighlighted = (component: AlloyComponent, hConfig: HighlightingConfig, h
 const getHighlighted = (component: AlloyComponent, hConfig: HighlightingConfig, _hState: Stateless): Optional<AlloyComponent> =>
   SelectorFind.descendant(component.element, '.' + hConfig.highlightClass).bind((e) => component.getSystem().getByDom(e).toOptional());
 
-const getByIndex = (component: AlloyComponent, hConfig: HighlightingConfig, hState: Stateless, index: number): Result<AlloyComponent, string> => {
+const getByIndex = (component: AlloyComponent, hConfig: HighlightingConfig, hState: Stateless, index: number): Result<AlloyComponent, Error> => {
   const items = SelectorFilter.descendants(component.element, '.' + hConfig.itemClass);
 
-  return Optional.from(items[index]).fold(() => Result.error<AlloyComponent, any>('No element found with index ' + index), component.getSystem().getByDom);
+  return Optional.from(items[index]).fold(() => Result.error(new Error('No element found with index ' + index)), component.getSystem().getByDom);
```

Comment:
This was incorrect as `component.getSystem().getByDom` returns `Result<AlloyComponent, Error>`.